Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -110,19 +100,46 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
     )
     .map { ($0.0, $0.1, $0.2, false) }
 
-    let newSetupIntentCards = self.newStripeIntentCreditCardProperty.signal.skipNil()
-      .map { data -> [PaymentSheetPaymentMethodCellData] in
-        let (displayData, setupIntent) = data
+    let newlyAddedCardProducer = self.newStripeIntentCreditCardProperty.signal.skipNil()
+      .switchMap { setupIntent, paymentMethodId -> SignalProducer<
+        Signal<UserCreditCards.CreditCard, ErrorEnvelope>.Event,
+        Never
+      > in
+        let input = CreatePaymentSourceSetupIntentInput.init(intentClientSecret: setupIntent, reuseable: true)
+        return AppEnvironment.current.apiService.addPaymentSheetPaymentSource(input: input)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .map { (envelope: CreatePaymentSourceEnvelope) -> UserCreditCards.CreditCard? in
+            guard envelope.createPaymentSource.isSuccessful else {
+              return nil
+            }
+
+            // There is a backend bug where the stripeCardId isn't being refreshed
+            // after the add card mutation is called. This adds it back in.
+            var card = envelope.createPaymentSource.paymentSource
+            if card.stripeCardId == nil {
+              card.stripeCardId = paymentMethodId
+            }
+
+            return card
+          }
+          .skipNil()
+          .materialize()
+      }
 
-        return [(
-          image: displayData.image,
-          redactedCardNumber: displayData.label,
-          clientSecret: setupIntent,
-          isSelected: false,
-          isEnabled: true
-        )]
+    let newCardsData: Signal<([PledgePaymentMethodCellData], UserCreditCards.CreditCard?), Never> = Signal
+      .combineLatest(
+        newlyAddedCardProducer.values(),
```

Comment:
I haven't added anything for `newlyAddedCardProducer.errors()`, not sure what to do with that. I guess display to the user?